diff --git a/ports/esp32/main.c b/ports/esp32/main.c
index 1420dd579..cefb1be1d 100644
--- a/ports/esp32/main.c
+++ b/ports/esp32/main.c
@@ -66,7 +66,7 @@
 
 // MicroPython runs as a task under FreeRTOS
 #define MP_TASK_PRIORITY        (ESP_TASK_PRIO_MIN + 1)
-#define MP_TASK_STACK_SIZE      (16 * 1024)
+#define MP_TASK_STACK_SIZE      (24 * 1024)
 
 // Set the margin for detecting stack overflow, depending on the CPU architecture.
 #if CONFIG_IDF_TARGET_ESP32C3
